src/ufuncs.c

```diff
@@ -1945,6 +1945,78 @@ static void polygonize_full_func(char** args, npy_intp* dimensions, npy_intp* st
 }
 static PyUFuncGenericFunction polygonize_full_funcs[1] = {&polygonize_full_func};
 
+static char nearest_points_dtypes[3] = {NPY_OBJECT, NPY_OBJECT, NPY_OBJECT};
+static void nearest_points_func(char** args, npy_intp* dimensions, npy_intp* steps,
+                                void* data) {
+  GEOSGeometry* in1 = NULL;
+  GEOSGeometry* in2 = NULL;
+  GEOSPreparedGeometry* in1_prepared = NULL;
+  GEOSGeometry** geom_arr;
+  GEOSCoordSequence* coord_seq = NULL;
+
+  CHECK_NO_INPLACE_OUTPUT(2);
+
+  // allocate a temporary array to store output GEOSGeometry objects
+  geom_arr = malloc(sizeof(void*) * dimensions[0]);
+  CHECK_ALLOC(geom_arr);
+
+  GEOS_INIT_THREADS;
+
+  BINARY_LOOP {
+    /* get the geometries: return on error */
+    if (!get_geom_with_prepared(*(GeometryObject**)ip1, &in1, &in1_prepared)) {
+      errstate = PGERR_NOT_A_GEOMETRY;
+      destroy_geom_arr(ctx, geom_arr, i - 1);
+      break;
+    }
+    if (!get_geom(*(GeometryObject**)ip2, &in2)) {
+      errstate = PGERR_NOT_A_GEOMETRY;
+      destroy_geom_arr(ctx, geom_arr, i - 1);
+      break;
+    }
+
+    if ((in1 == NULL) | (in2 == NULL) | GEOSisEmpty_r(ctx, in1) |
```

Comment:
Does somebody see something obviously wrong here?  The tests with `None` geometry input are segfaulting, and it's crashing inside `GEOSisEmpty`, so for some reason it's getting passed the ``(in1 == NULL) | (in2 == NULL)`` check.

Reply:
@brendan-ward if I write this again as two separate steps (checking for NULL and checking for empty), then it passes. So C does not short-circuit? 

-> Ah, I was using the bitwise `|` and not the logical `||` ... (and not only here, we do it in other places as well!). Properly using the logical one does short-circuit.

Reply:
Made a general PR to fix this -> https://github.com/pygeos/pygeos/pull/348